### DISCUSSION TITLE
Heat Pump Inverter Energy

### DISCUSSION THREAD

**Discussion by imcintyr** [PeteRager/lennoxs30#325] [General]
Hi all, I've been eyeing the "sensor.<name>_inverter_energy" for a few months now since its minimum number seems to be around 75W but mostly stays around 99W when the system is at rest. The blower is still moving air but the heat pump is not doing anything.

The diagnostic panel in the S40 thermostat says the INPUT V and A are 250 and .4 respectively which checks out. But the heat pump isn't actually doing anything and that is an input number so my assumption is that no electricity is actually being used. 

Is that correct? 

Ultimately I'd like to capture the number of kWh the unit uses but out-of-the-box that sensor would have nearly a 2.4 kWh bump in power use if this is the case.

There is a very strong chance I am not doing something correctly so please feel free to nudge me in the right direction!

Thanks!

**Comment by PeteRager** [comment]
The unit uses power even when it is not being used.  I have similar behavior on my outdoor air conditioning unit.  I turn the breaker off during the winter when I'm not using it to save power.

**Reply by imcintyr** [reply]
Thanks as always, Pete. You are very helpful!

For what it's worth, I did some more testing this morning and it looks like the furnace/HP system does draw 100W-ish but the sources are not just the HP inverter that I can tell. I couldn't isolate it. 

To give you an understanding of our environment, I have a TrueNAS server on which my HA instance runs. According to my Kill-a-Watt, that server draws between 95 and 100 W consistently after booting and settling down. WiFi and networking all run on a UPS so no draw from them as the UPS was unplugged for the testing. 

I turned off every breaker except for the server and the furnace. I was left with 135 to 140W for the power draw through HA's sensor on the Enphase (solar panel) system. When I threw the switch (not the breaker) on the furnace the T-stat turned off. I was left with 95-100W with just the server running, matching the measurements on the Kill-a-Watt; a delta of 40W comes from just the furnace power being 'ON'.

If I turn the continuous blower on to 625CFM, where it is normally, I see an increase in power consumption of about 70W, getting me near the 100W for the furnace in its lowest state. All the while, the inverter power stays close to 97 - 100W according to HA if the blower is on or not. 

I wasn't able to isolate the heat pump inverter specifically as drawing the 100W.

If you know of a way to isolate the inverter I would love to give it a test.

Thanks again!

**Comment by PeteRager** [comment]
I have an outdoor air conditioner unit and air handler inside.  There are two circuits for it a 120v and a 240v.  The 120 runs the air handler and the S30. I use the air handler in the winter to move wood stove heat around.  The 240v is for the air conditioner.  I also have CT sensors on the utility power coming into the breaker panel.

The CT sensors are rated for 200 amps, when there is very little power draw they are not super accurate. I think they claim 0.5% accuracy at full rating. So that's +- 1 amps or 120 watts.  I've verified this using a Fluke current meter.  I've thought about putting a second set of CT clamps on with a lower range like 0-20 and use this for lower power....or building a curve as the readings are repeatable.  Anyways...

The best way would be to put a meter directly on one of the  220v wires in the breaker panel going to heat pump.   As it's totally possible that Lennox has a CT sensor with a similar issue at low power. I believe they run some type of "heater" in there to keep the coolant within a range.  

I've wanted to create a power meter for the fan.  I think there is a diagnostic sensor that shows fan % of total power. But the actual power is dependent on the specific air handler model (as they have different size fans.) I've been unable to find a Lennox doc that describes it. 

Sounds like I should open the panel at somepoint and make these two measurement.

**Comment by PeteRager** [comment]
I opened the breaker panel today and put a meter on the 240v line going to the outdoor unit.

With the unit doing nothing, the meter reported 0.50 amps and the unit reported 0.40 amps
With the AC on, the meter reported 5.6 amps and the unit reported 5.6 amps.

I used the diagnostic sensor - sensor.sagehouse_ou_inverter_input_current (the integration uses this value * the ou reported voltage to calculate inverter power)

So the net is the power reporting from Lennox looks reasonably accurate.  The 1/10th of an amp difference could be due to power loss through the cable and/or meter accuracy at low currents.

I ran a similar test on the indoor unit.  I have a UV lamp in the indoor unit.  So I turned that off for these tests (l already have that on a smart switch). I have a CBA38MV-036-230-02 with an S30.

Idle = 0.23 amps
Fan at 2.5% power = 0.42 amps
Fan at 47% power = 3.24 amps
Fan at 0.5% power = 0.32 amps

in theory this should follow a straight line, I probably need a few more data points.

**Comment by imcintyr** [comment]
That's interesting, and thanks for the follow-up details.

I need to do one more isolation attempts with the HP inverter. Since I was fairly certain that I had it down to the blower, Reme Halo, and the S40, I need to take each one of those down to see how the values on the Enphase app and the HP inverter change if at all on the latter.

With the HP inverter, I am a little confused by the constant draw it shows in HA especially since the Enphase system doesn't seem to see it. It does see the HP clearly when the AC or Heat is running through it. I would think that if it sees the 5000W it draws when in heavy operation it would see the 95-100W when idle.

I'll get more data here in a week or so and let you know what I find.

**Comment by PeteRager** [comment]
I put the three measurements for the indoor unit into excel and plotted them, its a reasonably straight line.  This is for the particular model air handler I have.  Its a 30 ton.

https://tech.lennoxintl.com/C03e7o14l/VIu12Ch2uV/Corp1702a.pdf

Indoor Unit Amps = (0.07 * sensor.sagehouse_iu_indoor_blower_power) + 0.23

Certainly a few more measurements would give a more accurate constant.  But for now it's a good start for me.

That's on a 120v circuit so I'll multiply that by 1/2 of the sensor.sagehouse_ou_inverter_input_voltage to give watts.

**Reply by MNOiSY** [reply]
Aside from changing the diagnostic level to 2, is there something that needs to be done for the power sensor to work with the s40?

It worked fine with my S30's, but upgraded last week to the s40 and reset the config for the two new units. Local connection, different instance names, set level to 2 and still have no data from the power like the old one. Before I dive in, did I overlook something?

Thank you

**Reply by PeteRager** [reply]
There is a problem with the s40 where it only sends out diagnostics for an hour or so and then the diag level needs to be reset.  Give the script here a try:

https://github.com/PeteRager/lennoxs30/blob/master/docs/diagnostics.md

**Reply by MNOiSY** [reply]
Thanks for your reply, Pete.

The script is active, but the power sensor remains unchanged. The state is unknown and appears not to be receiving any data for either thermostat.

S40's are brand new, latest firmware, and all other sensors, parameters, and functions are working with the HA integration. Literally, the only thing not working (Aside from functions like rebooting them with the S30 method) is the power sensor. I have done multiple complete uninstallations and installations, ensured the diagnostic level is changed, tried the script, rebooted the device after it's connected, tried cloud connection to test, and cant seem to get the power data.

Side question, why does the script change the diagnostic level to 1, then 2 instead of conditionally checking if it's zero and updating it to 2 if true? I'm tired, so I might have overlooked it :)

Thanks again

**Reply by PeteRager** [reply]
Let's try this.

a) verify in the integration configuration that you have the inverter power sensor enabled, if not
b) reboot the thermostat, restart integration, set level to 2, if that doesn't fix
c) turn on message logging and debug logging for the integration. Instructions https://github.com/PeteRager/lennoxs30?tab=readme-ov-file#reporting-bugs , restart HA, post the message logging file and home-assistant.log file

**Reply by PeteRager** [reply]
The script was developed by a user with an S40 (I don't have one) and that was the pattern that worked.  The level would never return to zero, and oddly setting it to zero and then two would also not reenable it.

**Reply by MNOiSY** [reply]
Please find the logs for Home Assistant (HA) and each S40 thermostat at the following link:

[Logs Download Link](https://gofile.me/6Q9zv/IEwIkPnKs)

For testing purposes, I have included logs for scenarios where the internet was both enabled and disabled on the 2nd floor/Upstairs hardware. However, neither scenario appeared to affect the state or data of the inverter.

Thank you again

**Reply by PeteRager** [reply]
I'm not seeing any data related to the inverter in the message logs.  And hence while the sensor is being created there is no data for it to have a value.

Both systems are only reporting two pieces of equipment - the controller (AKA S40) and the furnace ("equipType": 16) so your outdoor unit which is either a heatpump (equipType 19) or an air conditioner i(equiptype 18) is not being reported.  What is the outdoor unit?  What do you see in the integrations devices?

Go into the S40 panel and look under the advanced settings you should be able to see a list of the equipment, perhaps it wasn't configured?

**Reply by MNOiSY** [reply]
I have two XC14 outdoor units. I do not remember what it showed in the thermostats, but, I will give it a look and post back. 

It is -10 Fahrenheit at the moment, so aside from turning them on quickly to test after installation, they haven't been used.

If the thermostat is in heat only mode, would that prevent the configuration from seeing it initially?

**Reply by PeteRager** [reply]
No, the equipment list is part of the config.  You could also try turning the breaker off or pulling the external disconnect on those units.  Someone else had the same issue a while ago. I'll see if I can find the issue. Have you ever had those unit works with the new S40?

**Reply by PeteRager** [reply]
#321 - this user had a similar problem with the outdoor unit not being detected.  I've asked how it was resolved,

**Reply by PeteRager** [reply]
Take a look at the response for #321
